index_api-v2.yaml

```diff
@@ -0,0 +1,722 @@
+                    status: "validation_failed"
+  /nodes:
+    post:
+      tags:
+        - Node Endpoints
+      summary: Add a node to the index
+      description: |
+        A node adds its profile to the index by posting the location of its profile (`profile_url`). The profile must include a list (`linked_schemas`) of one or more schemas against which the profile must be validated.
+
+        Each `linked_schemas` item is the name of a schema which can be found in the library using the following URL format: `{baseUrl}/schemas/{schema}.json`. For example:
+
+        `https://test-cdn.murmurations.network/schemas/test_schema-v1.json`
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/PostNodeRequest"
+            example:
+              profile_url: "https://somenode.org/optional-subdirectory/node-profile.json"
+      responses:
+        200:
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  data:
+                    $ref: "#/components/schemas/PostNodeResponse"
+              example:
+                data:
+                  node_id: "a55964aeaae9625dc2b8dbdb1c4ce0ed1e658483f44cf2be1a6479fe5e144d38"
+        400:
+          description: The profile_url is missing.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+              examples:
+                missing:
+                  value:
+                    message: "The profile_url parameter is missing."
+                    status: 400
+                invalid:
+                  value:
+                    message: "Invalid JSON body."
+                    status: 400
+        429:
+          $ref: "#/components/responses/TooManyRequests"
+        500:
+          $ref: "#/components/responses/ServerError"
+    get:
+      tags:
+        - Aggregator Endpoints
+      summary: Search for nodes
+      description: |
+        Aggregators can search for nodes based on any of the following parameters:
+
+        - A specific schema (`schema`)
+        - When the node was last validated by the index (`last_updated`)
+        - A distance in kilometers (_"25km"_) or miles (_"15mi"_)) from a specific geolocation (`lat`, `lon`, `range`)
+        - By city/town/village/etc, state/province/county/etc. and/or country. (`locality`, `region`, `country`)
+        - By node profile status (`posted` or `deleted`)
+        - By tags that describe the node (using an AND/OR filter with fuzzy or exact matching)
+        - By the nodes website address (`primary_url`)
+
+        Query results can be paginated using the `page` and `page_size` parameters.
+
+        Aggregators will eventually be required to authenticate themselves with a JWT to access this endpoint.
```

Comment:
Are we sure that JWT is the technology we want to use for this? I wonder if it's overkill for aggregator authentication (unless there's something we need to do that won't work with a simple API key approach).

Reply:
No, it was just a suggestion. I updated the wording to reflect both and that it is to be determined in the future.